packages/@storefront/sayt/src/autocomplete/index.ts

```diff
@@ -100,19 +106,29 @@ class Autocomplete {
     if (activate) {
       this.state.selected = index;
       if (indexExists) {
-        this.updateProducts(target, updateQuery);
+        const data = this.parseTarget(target);
+        if (updateQuery) this.updateQuery(data.query);
+        this.updateProducts(data);
```

Comment:
Is there a sane way to reduce the nesting here or collapse it in some way?

Reply:
Good idea. Will revisit.

Reply:
As discussed, there does not appear to be a way to refactor this without resulting in scope creep.

One of the aggravating factors is the manual addition/removal of a given CSS class. This logic may be removed if the component is updated to track both the 'selected' and 'currently hovered' elements within its own state. However, this update should be completed as part of a new ticket.